integration/src/test_util/capacity_checker.rs

```diff
@@ -0,0 +1,124 @@
+}
+
+pub fn run_circuit_capacity_checker(blocks: &[BlockTrace]) {
+    let each_tx_ccc_result = get_ccc_result_by_each_tx(blocks);
+
+    for light_mode in [true, false] {
+        let whole_block_ccc_result = get_ccc_result_by_whole_block(light_mode, blocks);
+        check_each_tx_and_whole_block_ccc_results(&each_tx_ccc_result, &whole_block_ccc_result);
+    }
+}
+
+fn get_ccc_result_by_each_tx(blocks: &[BlockTrace]) -> RowUsage {
+    log::info!(
+        "estimating circuit rows tx by tx, tx num {}",
+        blocks
+            .iter()
+            .map(|b| b.execution_results.len())
+            .sum::<usize>(),
+    );
+
+    let mut checker = CircuitCapacityChecker::new();
+    // checker.light_mode = false;
+
+    let start_time = std::time::Instant::now();
+
+    let mut tx_num = 0;
+    for (block_idx, block) in blocks.iter().enumerate() {
+        for tx_idx in 0..block.transactions.len() {
+            log::info!("processing {}th block {}th tx", block_idx, tx_idx);
+            #[rustfmt::skip]
+            /*  
+            The capacity_checker is expected to be run inside sequencer, where we don't have the traces of blocks, instead we only have traces of tx.
+            For the "tx_trace":
+                transactions: 
+                    the tx itself. For compatibility reasons, transactions is a vector of len 1 now.   
+                execution_results: 
+                    tx execution trace. Similar with above, it is also of len 1 vevtor.   
+                storage_trace: 
+                    storage_trace is prestate + siblings(or proofs) of touched storage_slots and accounts of this tx.
+            */
+
+            let tx_trace = BlockTrace {
+                transactions: vec![block.transactions[tx_idx].clone()],
+                execution_results: vec![block.execution_results[tx_idx].clone()],
+                storage_trace: block.tx_storage_trace[tx_idx].clone(),
+                chain_id: block.chain_id,
+                coinbase: block.coinbase.clone(),
+                header: block.header.clone(),
+                start_l1_queue_index: block.start_l1_queue_index,
+                tx_storage_trace: vec![], // not used
+            };
+            log::debug!("calling estimate_circuit_capacity");
+            let results = checker.estimate_circuit_capacity(&[tx_trace]);
+            log::info!(
+                "after {}th block {}th tx: {:#?}",
+                block_idx,
+                tx_idx,
+                results
+            );
+        }
+        tx_num += block.transactions.len();
+    }
+    log::info!("capacity_checker test done");
+    let ccc_result = checker.get_acc_row_usage(false);
+    log::info!(
+        "ccc result tx by tx {:#?}, after normalize {:#?}",
+        ccc_result,
+        ccc_result.normalize()
+    );
+
+    let avg_ccc_time = start_time.elapsed().as_millis() as usize / tx_num;
+    log::info!("avg time each tx: {avg_ccc_time}ms",);
+    assert!(avg_ccc_time < 100);
```

Comment:
@lispc could I also assert this time when runs in mock-testnet?

Reply:
no

Reply:
Fix to only assert `< 100ms` in test (not mock-testnet) in commit [abc5300](https://github.com/scroll-tech/scroll-prover/pull/269/commits/abc5300b8250363796bdbb971afe87a845543082).